java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -0,0 +1,253 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.se;
+
+import com.google.common.collect.Maps;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.InstanceOfTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.ParenthesizedTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TypeCastTree;
+import org.sonar.plugins.java.api.tree.UnaryExpressionTree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import javax.annotation.CheckForNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConstraintManager {
+
+  private final Map<Tree, SymbolicValue> map = new HashMap<>();
+  /**
+   * Map to handle lookup of fields.
+   * */
+  private final Map<Symbol, SymbolicValue> symbolMap = new HashMap<>();
+
+  public SymbolicValue createSymbolicValue(Tree syntaxNode) {
+    SymbolicValue result = map.get(syntaxNode);
+    if (result == null && syntaxNode.is(Tree.Kind.IDENTIFIER)) {
+      result = symbolMap.get(((IdentifierTree) syntaxNode).symbol());
+    } else if (result == null && syntaxNode.is(Tree.Kind.VARIABLE)) {
+      result = symbolMap.get(((VariableTree) syntaxNode).symbol());
+    }
+    if (result == null) {
+      result = new SymbolicValue.ObjectSymbolicValue(map.size() + ProgramState.EMPTY_STATE.constraints.size());
+      map.put(syntaxNode, result);
+      if (syntaxNode.is(Tree.Kind.IDENTIFIER)) {
+        symbolMap.put(((IdentifierTree) syntaxNode).symbol(), result);
+      } else if (syntaxNode.is(Tree.Kind.VARIABLE)) {
+        symbolMap.put(((VariableTree) syntaxNode).symbol(), result);
+      }
+    }
+    return result;
+  }
+
+  public SymbolicValue supersedeSymbolicValue(VariableTree variable) {
+    SymbolicValue result = new SymbolicValue.ObjectSymbolicValue(map.size() + ProgramState.EMPTY_STATE.constraints.size());
+    map.put(variable, result);
+    symbolMap.put(variable.symbol(), result);
+    return result;
+  }
+
+  public SymbolicValue eval(ProgramState programState, Tree node) {
+    Tree syntaxNode = skipTrivial(node);
+    switch (syntaxNode.kind()) {
+      case NULL_LITERAL:
+        return SymbolicValue.NULL_LITERAL;
+      case BOOLEAN_LITERAL:
+        boolean value = Boolean.parseBoolean(((LiteralTree) syntaxNode).value());
+        if (value) {
+          return SymbolicValue.TRUE_LITERAL;
+        }
+        return SymbolicValue.FALSE_LITERAL;
+      case VARIABLE:
+        SymbolicValue variableResult = programState.values.get(((VariableTree) syntaxNode).symbol());
+        if (variableResult != null) {
+          // symbolic value associated with local variable
+          return variableResult;
+        }
+        break;
+      case IDENTIFIER:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)